### PR TITLE
DBZ-3722 MySQL error handler add relevant exceptions

### PR DESCRIPTION
**Pull request by wingerx**
https://issues.redhat.com/browse/DBZ-3722

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlErrorHandler.java [hunk 1]
```diff
@@ -39,6 +41,9 @@ else if (throwable instanceof ServerException) {
         else if (throwable instanceof DebeziumException && throwable.getCause() != null) {
             return isRetriable(throwable.getCause());
         }
+        else if (throwable instanceof SocketException || throwable instanceof EOFException) {
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlErrorHandler.java:44]: Could we perhaps further narrow down that condition, like a status code or (parts of) the error message?

Comment (wingerx) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlErrorHandler.java:44]: I tracked the exception stack of the code, BinaryLogClient does not throw a wrapped exception but io EOFExcetion or network SocketException. So... any other good ideas?

**Comment by gunnarmorling** [comment]
So it throws those exceptions without any message at all?

>

**Comment by gunnarmorling** [comment]
Looking ath the stacktrace you shared in the issue, there's e.g. this message:

> Network is unreachable: recv failed

I'd feel more comfortable if we could narrow down the restart cases based on that.

**Comment by wingerx** [comment]
> So it throws those exceptions without any message at all?

Yes, I think so.
I copy some  codes from `BinaryLogClient.java`
```java
                try {
                    // ... ignore
                } catch (Exception e) {
                    Throwable cause = e instanceof EventDataDeserializationException ? e.getCause() : e;
                    if (cause instanceof EOFException || cause instanceof SocketException) {
                        throw e;
                    }
                    if (isConnected()) {
                        for (LifecycleListener lifecycleListener : lifecycleListeners) {
                            lifecycleListener.onEventDeserializationFailure(this, e);
                        }
                    }
                    continue;
                }
```
or modify `wrap` method in `MySqlStreamingChangeEventSource.java`. But i donot think this has any difference exactly.

**Comment by jpechane** [comment]
@wingerx Yes, as @gunnarmorling has pointed out. There is a chain of cause exeptions and one of them is `SocketInputStream`. Maybe the PR should be updated to go through the chain and check this concrete expection together with a substring of the exception message.

**Comment by wingerx** [comment]
Sorry, I did not reproduce this problem, I will close this pr and resubmit when has a better plan

**Pull request closed without merging**